test/ha/ha.go

```diff
@@ -23,12 +23,79 @@ import (
 	"time"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"knative.dev/pkg/test"
 	"knative.dev/pkg/test/logging"
 )
 
-// WaitForNewLeader waits until the holder of the given lease is different from the previousLeader.
+// GetLeaders collects all of the leader pods from the specified deployment.
+func GetLeaders(client *test.KubeClient, deploymentName, namespace string) ([]string, error) {
+	leases, err := client.Kube.CoordinationV1().Leases(namespace).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("error getting leases for deployment %q: %w", deploymentName, err)
+	}
+	var pods []string
+	for _, lease := range leases.Items {
+		if lease.Spec.HolderIdentity == nil {
+			continue
+		}
+		pod := strings.Split(*lease.Spec.HolderIdentity, "_")[0]
+
+		// Deconstruct the pod name and look for the deployment.  This won't work for very long deployment names.
+		parts := strings.Split(pod, "-")
+		if len(parts) < 3 {
+			continue
+		}
+		if strings.Join(parts[:len(parts)-2], "-") != deploymentName {
+			continue
+		}
```

Comment:
I wonder if we can use one split/find call?

Reply:
🤷

Reply:
😢